### PR TITLE
Map-ready TFR display projection (shared vertical unions)

### PR DESCRIPTION
**Pull request by alexwitherspoon**
Closes #270.

Part of epic #245. Stacks on #269 (`feat/airspace-platform-250`).

## Summary
- Add a server-side **map-ready display projection** (`lib/notam/map-display-projection.php`) applied when building `/api/notam-map.php` GeoJSON.
- Overlapping polygons that share restriction kind, map style bucket, and vertical limits (for example `SFC-9000`) collapse into one convex-hull footprint.
- Unified airspace store is unchanged; merged features expose `display_merged`, `member_notam_ids`, and `member_count`.
- Directory map popup lists overlapping NOTAM ids for research; circles and different vertical bands stay separate.
- Document the store-vs-display split in `docs/API.md` and `docs/ARCHITECTURE.md`.

## Test plan
- [x] `PHPUNIT_ARGS='--filter NotamMapDisplayProjectionTest' make test-unit`
- [x] Related map/airspace unit filters
- [x] `make test-ci`
- [x] After merge of lower stack PRs, confirm eastern Oregon fire TFRs render fewer stacked fills on local map
- [x] Confirm popup shows member NOTAM list on a merged footprint

### CODE REVIEW

**Comment by alexwitherspoon** [comment]
Follow-up polish on this PR: TFR headline categories.

Brownsville / Boca Chica standing SSI (`49 USC 40103`, `14 CFR 99.7`, NTL defense airspace) was labeled **VIP TFR** because SSI UAS exception text contains "DISASTER RESPONSE". Classification now recognizes a **Security TFR** bucket for those markers and no longer treats bare "DISASTER" as VIP. Dedicated rocket/space-launch wording still maps to Rocket test TFR; this facility SSI is Security, which matches the map signal we want.

**Comment by alexwitherspoon** [comment]
Follow-up: classification edge-case pass from the local-data audit.

- Presidential rings citing `91.141` (Thurmont / Bedminster / Morristown) → **VIP TFR**, even when SSI/`40103` is also present
- Facility SSI without `91.141` (Brownsville / Boca Chica) stays **Security TFR**
- `91.145` aerial demonstration boilerplate → **Airshow TFR** (not Sporting event)
- Rocket launch vs static engine test → **Space launch TFR** / **Rocket test TFR**
- `49 USC 44812` large gatherings → **Event UAS TFR**
- Non-TFR prose no longer fail-opens to the label "TFR"
- Guardrail/tests keep one-word SSI `FIREFIGHTING` from becoming Fire

File: lib/notam/map-display-projection.php [hunk 1]
```diff
@@ -0,0 +1,956 @@
...
+    $props = is_array($feature['properties'] ?? null) ? $feature['properties'] : [];
+    $headline = (string) ($props['banner_headline'] ?? '');
+    if (preg_match('/\b(SFC|\d+)\s*-\s*(\d+)\s*ft\b/i', $headline, $m) === 1) {
+        $lo = strtoupper($m[1]);
+
+        return $lo . '-' . $m[2];
+    }
+
+    return null;
+}
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply lib/notam/map-display-projection.php]: Good catch. The vertical key dropped AGL/MSL (and FL/UNL) after `ft`, so same altitudes with different datums could share a merge bucket. Fixed in 51010945: include the datum suffix in the key, bump display projection to v5, and add unit coverage for AGL vs MSL.
